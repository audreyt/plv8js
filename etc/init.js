var module, require;
module = function(pathfilename){
  this.id = 'module_' + (new Date).getTime();
  this.pathfilename = pathfilename;
  return this.exports = {};
};
require = function(modulename){
  var filename, delim, c, prefixes, cwd, path_delim, dirar, i$, len$, prefix, rp, stats, fname, packageJson, __dirname, __filename, T, pos, m, body, fn, oldCwd;
  filename = modulename;
  delim = require.path_delim;
  c = filename.charAt(0);
  prefixes = ['/usr/local/plv8/plv8_modules/', '/usr/local/plv8/lib/'];
  if (c == '.' || c == '/') {
    prefixes = [''];
  }
  if (c != '.' && c != '/') {
    cwd = native_fs_.getcwd();
    path_delim = require.path_delim;
    dirar = cwd.split(path_delim);
    do {
      if (native_fs_.statSync(dirar.join(path_delim) + '/package.json').isFile) {
        prefixes.push((dirar.concat(['node_modules', ''])).join(path_delim));
      }
    } while (dirar.pop());
  }
  for (i$ = 0, len$ = prefixes.length; i$ < len$; ++i$) {
    prefix = prefixes[i$];
    rp = require.resolvePath(prefix + "" + filename);
    if (require.loaded[rp] != null) {
      return require.loaded[rp].exports;
    }
    if (require.loading[rp] != null) {
      return require.loading[rp].exports;
    }
    stats = native_fs_.statSync(rp);
    switch (false) {
    case !stats.isDirectory:
      fname = rp + "" + delim + 'package.json';
      packageJson = new Function("return " + native_fs_.readFileSync(fname))();
      if (packageJson.main == null) {
        throw new Error("cannot find module of " + modulename + " (" + fname + " " + packageJson + "})");
      }
      __dirname = rp;
      __filename = packageJson.main;
      T = rp + "/" + delim + __filename;
      if (native_fs_.statSync(T).isDirectory) {
        T += '/index.js';
      } else {
        if (!/\.js$/.test(T)) {
          T += '.js';
        }
      }
      break;
    case !stats.isFile:
      T = rp;
      break;
    case !native_fs_.existsSync(rp + ".js"):
      T = rp + ".js";
      break;
    case !native_fs_.existsSync(rp + ".node"):
      T = rp + ".node";
      break;
    case this["native_" + modulename + "_"] == null:
      return this["native_" + modulename + "_"];
    default:
      continue;
    }
    pos = T.lastIndexOf(delim);
    __dirname = T.substring(0, pos);
    __filename = T.substring(pos + 1);
    switch (__filename.substring(__filename.lastIndexOf('.'))) {
    case '.js':
      m = new module(T);
      require.loading[rp] = m;
      body = native_fs_.readFileSync(T);
      fn = new Function('module', 'exports', '__dirname', '__filename', body);
      oldCwd = native_fs_.getcwd();
      native_fs_.chdir(__dirname);
      m.exports = {};
      fn(m, m.exports, __dirname, __filename);
      require.loaded[rp] = m;
      require.loading[rp] = void 8;
      native_fs_.chdir(oldCwd);
      return m.exports;
    case '.node':
      m = new module(T);
      native_fs_.loadso(T, m.exports);
      require.loaded[T] = m;
      return m.exports;
    default:
      continue;
    }
  }
  throw new Error("cannot find module of " + modulename);
};
require.resolvePath = function(filename){
  var path_delim, dir, dirar, i$, ref$, len$, t;
  path_delim = require.path_delim;
  if (filename[0] === path_delim) {
    return filename;
  }
  dir = typeof __dirname != 'undefined' && __dirname !== null
    ? __dirname
    : native_fs_.getcwd();
  dirar = dir.split(path_delim);
  for (i$ = 0, len$ = (ref$ = filename.split(path_delim)).length; i$ < len$; ++i$) {
    t = ref$[i$];
    switch (t) {
    case '.':
      continue;
    case '..':
      dirar.pop();
      break;
    default:
      dirar.push(t);
    }
  }
  return dirar.join(path_delim);
};
require.path_delim = '/';
require.loading = {};
require.loaded = {};