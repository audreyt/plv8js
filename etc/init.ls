module = (pathfilename) ->
  @id = 'module_' + (new Date).getTime!
  @pathfilename = pathfilename
  @exports = {}

require = (modulename) ->
  filename = modulename
  delim = require.path_delim
  c = filename.charAt 0
  filename = './node_modules/' + filename if c isnt '.' and c isnt '/' and (filename.indexOf delim) < 0
  rp = require.resolvePath filename
  if require.loaded[rp] isnt ``undefined`` then return require.loaded[rp].exports else if require.loading[rp] isnt ``undefined`` then return require.loading[rp].exports
  __dirname = void
  __filename = void
  T = void
  stats = native_fs_.statSync rp
  if stats.isDirectory
    packagejson = new Function 'return ' + native_fs_.readFileSync rp + delim + 'package.json'
    throw new Error 'cannot find module of ' + modulename if packagejson.main is ``undefined``
    __dirname = rp
    __filename = packagejson.main
    T = rp + delim + __filename
  else
    if stats.isFile
      T = rp
    else
      if native_fs_.existsSync rp + '.js'
        T = rp + '.js'
      else
        if native_fs_.existsSync rp + '.node'
          T = rp + '.node'
        else
          if typeof @['native_' + modulename + '_'] isnt 'undefined' then return @['native_' + modulename + '_'] else throw new Error 'cannot find module of ' + modulename
  pos = void
  pos = T.lastIndexOf delim
  __dirname = T.substring 0, pos
  __filename = T.substring pos + 1
  suffix = __filename.substring __filename.lastIndexOf '.'
  if suffix is '.js'
    m = new module T
    require.loading[rp] = m
    fn = new Function 'module, exports, __dirname, __filename', native_fs_.readFileSync T
    exports = m.exports
    fn m, exports, __dirname, __filename
    require.loaded[rp] = m
    require.loading[rp] = ``undefined``
    for key of exports
      m.exports[key] = exports[key]
    m.exports
  else
    if suffix is '.node'
      m = new module T
      native_fs_.loadso T, m.exports
      require.loaded[T] = m
      m.exports

require.resolvePath = (filename) ->
  path_delim = require.path_delim
  dirar = void
  if filename.0 is path_delim
    filename
  else
    fnar = filename.split path_delim
    dir = void
    if typeof __dirname is 'undefined' then dir = native_fs_.getcwd! else dir = __dirname
    dirar = dir.split path_delim
    i = 0
    n = fnar.length
    while i < n
      t = fnar[i]
      if t is '.' then continue else if t is '..' then dirar.pop! else dirar.push t
      i++
    dirar.join path_delim

require.path_delim = '/'

require.loading = {}

require.loaded = {}
