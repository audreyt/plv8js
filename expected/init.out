-- INSTALL
SET client_min_messages = warning;
\set ECHO none
RESET client_min_messages;
CREATE LANGUAGE plv8;
